### ISSUE TITLE
extension-pkg-allow-list seems to be ignored

### ISSUE THREAD

**Issue by Conkey100** [PyCQA/pylint#10662]
### Bug description

All of the directives in the .pylintrc file are honoured except  extension-pkg-allow-list.

```python
# Example
from wx import MessageBox
```

### Configuration

```ini
[MAIN]

extension-pkg-allow-list=wx
```

### Command used

```shell
It is built into my IDE (Spyder v6.1)
```

### Pylint output

```python
No name 'MessageBox' in module 'wx'
```

### Expected behavior

It should read the wx module and only flag miss-spelled, incorrect, etc. members

### Pylint version

```shell
pylint 3.3.8
astroid 3.3.11
Python 3.12.11 | packaged by conda-forge | (main, Jun  4 2025, 14:45:31) [GCC 13.3.0]
```

### OS / Environment

Mint 22.1 Cinnamon 

### Additional dependencies

```python

```

**Comment by DanielNoord** [comment]
It could also be that even with the allow list we still have trouble understanding the package. Did this work on an earlier version?

Also note that you are not using the latest version of `pylint`, so updating is likely better.

**Comment by Conkey100** [comment]
I haven't used pylint before, so I can't comment directly on whether it was working earlier. When I noticed the issue (every line using wxPython was flagged as an error) , I did a search and found several posts on stack overflow and the wxPython forums stating that everything worked as expected after adding it to the allow list. [Here is an example](https://stackoverflow.com/questions/65351051/vscode-wxpython-package-intellisense-module-wx-has-no-frame-member)

I'm unfortunately not able to upgrade pylint as it is built into the IDE.

When I have time in a couple of days, I will try with a stand-alone version.

**Comment by Conkey100** [comment]
I installed pylint and tested outside of the IDE (both 3.3.8 and the llatest version). It seems to work as expected - I tested the same file and none of the lines using wxPython were flagged.

**Comment by Pierre-Sassoulas** [comment]
Probably a downstream configuration issue in the IDE then. Thank you for the update.

**Comment by marcelteun** [comment]
I had the same problem with wx. In my case  I could fix this by using:

[MASTER]                                                                                                                                                                                  
ignored-modules=wx 

instead
